### PR TITLE
feat: Introduce the Cart Library

### LINKED ISSUE #13044: [Epic] Introduce the Cart Library
Move the cart and order related code to one or 2 feature libs

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts [hunk 1]
```diff
@@ -19,11 +24,16 @@ export class PageLayoutService {
...
     private config: LayoutConfig,
     private breakpointService: BreakpointService,
-    @Optional()
-    @Inject(PAGE_LAYOUT_HANDLER)
-    private handlers: PageLayoutHandler[]
+    // TODO: need doc this in 5.0
+    //@Optional()
+    //@Inject(PAGE_LAYOUT_HANDLER)
+    //private handlers: PageLayoutHandler[],
+    private unifiedInjector: UnifiedInjector
```
Comment (plabadie) [inline projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts:27]: What do we do with this?

File: integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-method.component.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
-import { DP_CARD_REGISTRATION_STATUS } from '../../../utils/dp-constants';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
```
Comment (plabadie) [inline integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-method.component.ts:1]: Lets review changes to dp

File: feature-libs/cart/base/components/cart-shared/cart-item/cart-item.component.html [hunk 3]
```diff
@@ -88,7 +88,7 @@
               [readonly]="
                 !item.updateable || readonly || options.isSaveForLater
               "
-              [max]="item.product.stock?.stockLevel"
+              [max]="item.product?.stock?.stockLevel || 1000"
```
Comment (plabadie) [inline feature-libs/cart/base/components/cart-shared/cart-item/cart-item.component.html]: Let's review this

File: feature-libs/cart/karma.conf.js [hunk 4]
```diff
@@ -4,32 +4,30 @@
...
       reporters: [{ type: 'lcov', subdir: '.' }, { type: 'text-summary' }],
       check: {
         global: {
-          statements: 80,
-          lines: 80,
-          branches: 68,
-          functions: 80,
+          statements: 90,
+          lines: 90,
+          branches: 75,
```
Comment (plabadie) [inline feature-libs/cart/karma.conf.js:29]: lets review this

File: feature-libs/cart/base/root/components/add-to-cart/add-to-cart.component.ts [hunk 5]
```diff
@@ -138,40 +126,43 @@ export class AddToCartComponent implements OnInit, OnDestroy {
...
     this.activeCartService
       .getEntries()
       .pipe(take(1))
-      .subscribe((entries) => {
-        this.numberOfEntriesBeforeAdd = entries.length;
-        this.openModal();
+      .subscribe((cartEntries) => {
         this.activeCartService.addEntry(this.productCode, quantity);
+
+        // A CartUiEventAddToCart is dispatched.  Thiis event is intended for the UI
```
Comment (plabadie) [inline feature-libs/cart/base/root/components/add-to-cart/add-to-cart.component.ts]: typo

File: projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts [hunk 6]
```diff
@@ -19,11 +24,16 @@ export class PageLayoutService {
     private cms: CmsService,
     private config: LayoutConfig,
     private breakpointService: BreakpointService,
-    @Optional()
-    @Inject(PAGE_LAYOUT_HANDLER)
-    private handlers: PageLayoutHandler[]
+    // TODO: need doc this in 5.0
```
Comment (plabadie) [inline projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts]: Lets review this

File: projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts:27]: created followup ticket GH-14918

File: integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-method.component.ts (hunk 2 above)
Comment (plabadie) [reply integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-method.component.ts:1]: Reviewed

File: feature-libs/cart/base/components/cart-shared/cart-item/cart-item.component.html (hunk 3 above)
Comment (plabadie) [reply feature-libs/cart/base/components/cart-shared/cart-item/cart-item.component.html]: Removing the default value.  It was removed a while ago on develop, but had not been ported to the cart-lib file.
https://github.com/SAP/spartacus/pull/14086/files

File: feature-libs/cart/karma.conf.js (hunk 4 above)
Comment (plabadie) [reply feature-libs/cart/karma.conf.js:29]: We keep it like this.

File: feature-libs/cart/base/root/components/add-to-cart/add-to-cart.component.ts (hunk 5 above)
Comment (plabadie) [reply feature-libs/cart/base/root/components/add-to-cart/add-to-cart.component.ts]: fixed

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**